scene/gui/rich_text_effect.cpp

```diff
@@ -0,0 +1,90 @@
+#include "rich_text_effect.h"
+
+#include "core/script_language.h"
+
+void RichTextEffect::_bind_methods() {
+	BIND_VMETHOD(MethodInfo(Variant::INT, "_process_custom_fx", PropertyInfo(Variant::OBJECT, "char_fx", PROPERTY_HINT_RESOURCE_TYPE, "CustomFXChar")));
+}
+
+Variant RichTextEffect::get_bbcode() const {
+	Variant r;
+	if (get_script_instance()) {
+		if (!get_script_instance()->get("bbcode", r)) {
+			String path = get_script_instance()->get_script()->get_path();
+			r = path.get_file().get_basename();
+		}
+	}
+	return r;
+}
+
+bool RichTextEffect::_process_effect_impl(Ref<CharFXTransform> cfx) {
+	Variant v = false;
```

Comment:
That's a bit weird, it would be better to make this a proper `bool`, and defer the casting `Variant` to `bool` when parsing the return value of `call()`.

Reply:
So I've resolved this too, but I'm going to keep it unresolved for now since I haven't actually testing this thoroughly enough with my test project.